src/df.jl

```diff
@@ -45,18 +60,23 @@ end
 
 stringify(x) = filter(t -> t != ':', string(x))
 
-select_column(df, s) = haskey(df, s) ? convert_column(df[s]) : s
-
-convert_column(col) = col
+compute_name(df, i) = column_names(IterableTables.getiterator(df))[i]
 
-function convert_column(col::AbstractDataArray{T}) where T
-    try
-        convert(Array, col)
-    catch
-        error("Missing data of type $T is not supported")
+function compute_all(df, syms...)
+    iter = IterableTables.getiterator(df)
```

Comment:
You probably want to check whether you actually got an iterable table by calling ``TableTraits.isiterabletable(df)``, right? Someone could have passed you something that is something else entirely.